### PR TITLE
Handle "new-style" COSE envelop and validate transparent receipts

### PR DESCRIPTION
**Pull request by micromaomao**
Prepare for validating transparant fragments.

```
> sign1util print -in ~/cplat-0.2.14-dev.202603261649/package/LinuxBootFiles/transparent_reference_info.cose
checkCoseSign1 passed
CCF receipt 0 from esrp-cts-db.confidential-ledger.azure.com validation passed
iss: did:x509:0:sha256:I__iuL25oXEVFdTP_aBLx_eT1RPHbCQ_ECBQfYZpt9s::eku:1.3.6.1.4.1.311.76.59.1.2
feed: ContainerPlat-AMD-UVM
cty: application/json
pubkey: MIIBojANBgkqhkiG9w0BAQEFAAOCAY8AMIIBigKCAYEAxTSM4xP4oW53VIo0Q0rnOk99dFCjkrEoDr0Euwb9KHvMHGk30wkA62B5ey0MHtL6zwNCiLdJxatExQrm0xC8bv3q5fbBjGZ0yZGIMlTKrbI7zHz/ZF2pm0ojqSWlV2fcaEgM4UuyK4waxZPXm/Z37oGVdHSOJH0wNn1VqLIcfO6mhxq+rqUaKwNNjLIkmMyClTQKmRIAeB9bSd/TJlsHfaOCbodXKIjyiqqSZsBf5GFQDMKS+j3CzEPFIMg6eK0nADEEtygnq2wU0tT/yOL6F/nCRAExYulJ97rq/X2J1IERKQSTIuGQ7eCIL0j0rVLDoi9rq+Lf4vYUCayeKDnHmjJ6DmwPeXdQmoZftSvWLkFXdzkhT3ue8O5IIM8z0VkoVH2mgWeqsHuhdyJCuyELk4A33+Yo+OD+JLTNMeGLas7g8mKAbK6HnPf47v90ofh7BVytqpuugKvjei+UU46tgmj2jvbZuyj5j83V/Jha9yfyJquv0mA0G3LhDB4W8hrZAgMBAAE=
pubcert: MIIGazCCBFOgAwIBAgITMwAAAE+CVvVEUdIgyAAAAAAATzANBgkqhkiG9w0BAQwFADBVMQswCQYDVQQGEwJVUzEeMBwGA1UEChMVTWljcm9zb2Z0IENvcnBvcmF0aW9uMSYwJAYDVQQDEx1NaWNyb3NvZnQgU0NEIFByb2R1Y3RzIFJTQSBDQTAeFw0yNTA1MTUxODU3MDNaFw0yNjA1MTUxODU3MDNaMGwxCzAJBgNVBAYTAlVTMRMwEQYDVQQIEwpXYXNoaW5ndG9uMRAwDgYDVQQHEwdSZWRtb25kMR4wHAYDVQQKExVNaWNyb3NvZnQgQ29ycG9yYXRpb24xFjAUBgNVBAMTDUNvbnRhaW5lclBsYXQwggGiMA0GCSqGSIb3DQEBAQUAA4IBjwAwggGKAoIBgQDFNIzjE/ihbndUijRDSuc6T310UKOSsSgOvQS7Bv0oe8wcaTfTCQDrYHl7LQwe0vrPA0KIt0nFq0TFCubTELxu/erl9sGMZnTJkYgyVMqtsjvMfP9kXambSiOpJaVXZ9xoSAzhS7IrjBrFk9eb9nfugZV0dI4kfTA2fVWoshx87qaHGr6upRorA02MsiSYzIKVNAqZEgB4H1tJ39MmWwd9o4Juh1coiPKKqpJmwF/kYVAMwpL6PcLMQ8UgyDp4rScAMQS3KCerbBTS1P/I4voX+cJEATFi6Un3uur9fYnUgREpBJMi4ZDt4IgvSPStUsOiL2ur4t/i9hQJrJ4oOceaMnoObA95d1Cahl+1K9YuQVd3OSFPe57w7kggzzPRWShUfaaBZ6qwe6F3IkK7IQuTgDff5ij44P4ktM0x4YtqzuDyYoBsroec9/ju/3Sh+HsFXK2qm66Aq+N6L5RTjq2CaPaO9tm7KPmPzdX8mFr3J/Imq6/SYDQbcuEMHhbyGtkCAwEAAaOCAZswggGXMA4GA1UdDwEB/wQEAwIHgDAjBgNVHSUEHDAaBgsrBgEEAYI3TDsBAQYLKwYBBAGCN0w7AQIwHQYDVR0OBBYEFKEPDqVK/oBtc4H6FxKDddVcgjfGMEUGA1UdEQQ+MDykOjA4MR4wHAYDVQQLExVNaWNyb3NvZnQgQ29ycG9yYXRpb24xFjAUBgNVBAUTDTQ3Mjk3Mis1MDQ4NjcwHwYDVR0jBBgwFoAUVc1NhW7NSjXDjj9yAbqqmBmXS6cwXgYDVR0fBFcwVTBToFGgT4ZNaHR0cDovL3d3dy5taWNyb3NvZnQuY29tL3BraW9wcy9jcmwvTWljcm9zb2Z0JTIwU0NEJTIwUHJvZHVjdHMlMjBSU0ElMjBDQS5jcmwwawYIKwYBBQUHAQEEXzBdMFsGCCsGAQUFBzAChk9odHRwOi8vd3d3Lm1pY3Jvc29mdC5jb20vcGtpb3BzL2NlcnRzL01pY3Jvc29mdCUyMFNDRCUyMFByb2R1Y3RzJTIwUlNBJTIwQ0EuY3J0MAwGA1UdEwEB/wQCMAAwDQYJKoZIhvcNAQEMBQADggIBAIFlKzk+6/SVwScu6Am5rcBiJpjxrvGPG9BKhS4Vl3zVMw/oxhMahHpL3VB4n+HjibYakHISn76jbYIeE9rwRlfhCazzY3J1wHd0Cqx+22+Vp8JGfpXGYN606yy+6Oycvdt5+lHX5OKA82XDKGkMVr3JuVH1NLXuMy4HbJgoDt9o1YDrQUsuLIiyxKIo6jtdzWL+X54EQ47aUtQ23LX1YxXRzj1UdO7y0hCmwj8qIol0OXQQP2UkwxPhRrCg983BWtrzKPHZIM0S8MU9EKpklWvLnrOp55bmJenUGPakmbLJXke18A38wba+TJy0icxumAQ0qhaIxqO3XAeI8PGc5cRM8fyiTbcr0ioSh3He0Xrtf61AwQB1r2rsmPDQiV3f209qGzaOmlEoPtZVaZrx3bR2pvtITNlSmGwTRU9rONukctvVn8ymUL7zLohMTWHYwNHbV6pWY9j7MlHDz1SY6rncpg7asjdnjYAOI2D8832Jrxbn5d3hwaoEz6t7hsDPUeKgXROe5+UcP8JCkSaB1UtpfZ0eyE/6ebqXZ1UcxPEVyHq1SyM+SK3YqglHFxzJZCle/rIaw6AzolCqcuy0NF1EP3FcSAzZtjOenctMvDr/oh8RNfIyG96kCJDncJ/a6hYiV0usAoL+N93ikzUhq2ZekShUPgyGf6EFg1PH3iih
all protected headers:
  1: PS384
  3: "application/json"
  15: {1: "did:x509:0:sha256:I__iuL25oXEVFdTP_aBLx_eT1RPHbCQ_ECBQfYZpt9s::eku:1.3.6.1.4.1.311.76.59.1.2", 2: "ContainerPlat-AMD-UVM", 6: 2026-03-26 17:01:36 +0000 UTC, "svn": 104}
  33: ...
  34: ...
all unprotected headers:
  394: ...
payload:
{
  "x-ms-sevsnpvm-launchmeasurement": "dc3f5a934489232a9b1818f12a0a88d2324ced00f8ab370f40451a76b7880bc3e211849a0739642d3d6c3b2b4bfb9866"
}

receipt 0:
  protected headers:
    ccf.v1: {"txid": "685.17237"}
    1: ES384
    4: "da7694f16def5a056ca96afb21e89a9450e4cc875e2de351da76d99544a3e849"
    395: 2
    15: {6: 1774544498, 1: "esrp-cts-db.confidential-ledger.azure.com", 2: "scitt.ccf.signature.v1"}
  unprotected headers:
    396:
      -1: inclusion proof
  payload: ""
```

Todo:
- Mayve we should keep the existing output of `sign1util print`, and only prints the headers and receipts if --print-headers / --print-receipts flags are provided?

### CODE REVIEW

**Review by KenGordon — DISMISSED** [review]
Maybe break the key fetch out of main.

**Comment by micromaomao** [comment]
need to check that receipt actually matches the data

File: cmd/sign1util/ccf_keyfetch.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	yBytes, err := base64.RawURLEncoding.DecodeString(k.Y)
+	if err != nil {
+		return nil, fmt.Errorf("decoding y: %w", err)
+	}
+	return &ecdsa.PublicKey{
+		Curve: curve,
+		X:     new(big.Int).SetBytes(xBytes),
+		Y:     new(big.Int).SetBytes(yBytes),
+	}, nil
+}
```
[the remark this answers is not in the document]
Comment (micromaomao) [reply cmd/sign1util/ccf_keyfetch.go:65]: todo

File: cmd/sign1util/ccf_keyfetch.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+// error aborts the connection.
+type CertVerifier func(issuer string, cert *x509.Certificate) error
+
+// fetchIssuerJWKS GETs https://<issuer>/jwks and returns the keys keyed by
+// their `kid`. If verifyCert is not nil, the leaf certificate presented by the
+// server is passed to verifyCert.
+func fetchIssuerJWKS(issuer string, verifyCert CertVerifier) (map[string]crypto.PublicKey, error) {
+	url := "https://" + issuer + "/jwks"
+	tlsConfig := &tls.Config{InsecureSkipVerify: true} //nolint:gosec // CCF uses self-signed certs that are supposed to be validated via attestation, and so will never pass the normal verification.
+	if verifyCert != nil {
```
[the remark this answers is not in the document]
Comment (micromaomao) [reply cmd/sign1util/ccf_keyfetch.go:117]: Todo:
We should make it so that by default we have a domain allow list that only allows subdomains of "confidential-ledger.azure.com" for this key fetching, but the user can add to this list via a new `--allow-jwks-domain` argument for `sign1util check` / `sign1util print`

File: cmd/sign1util/main.go [hunk 3]
```diff
@@ -33,13 +81,57 @@ func checkCoseSign1(inputFilename string, chainFilename string, didString string
...
+			return nil, fmt.Errorf("fetching CCF receipt keys: %w", err)
+		}
+		for i, r := range unpacked.Receipts {
+			if err := r.Validate(receiptKeys); err != nil {
+				fmt.Fprintf(os.Stdout, "CCF receipt %d from %s validation failed - %s\n", i, r.Issuer, err)
+				return nil, fmt.Errorf("CCF receipt %d from %s validation failed: %w", i, r.Issuer, err)
+			}
+			fmt.Fprintf(os.Stdout, "CCF receipt %d from %s validation passed\n", i, r.Issuer)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (micromaomao) [reply cmd/sign1util/main.go]: todo add --validate-receipts opt-in to `sign1util check` and `sign1util print`

File: cmd/sign1util/main.go [hunk 4]
```diff
@@ -1,15 +1,63 @@
...
+// CCF nodes serve TLS using a self-signed certificate whose authenticity is
+// backed by attestation rather than a public CA. Since we have no way to
+// validate the CCF's attestation evidence here, we simply print summary details
+// of a CCF node's TLS certificate and unconditionally accepts it, which is
+// acceptable for this tool.
+func acceptAndPrintCert(issuer string, cert *x509.Certificate) error {
+	fp := sha256.Sum256(cert.Raw)
+	fmt.Fprintf(os.Stdout, "%s: accepting TLS certificate subject=%q issuer=%q notAfter=%s sha256=%s\n",
+		issuer, cert.Subject, cert.Issuer, cert.NotAfter.Format("2006-01-02"), hex.EncodeToString(fp[:]))
+	return nil
```
[the remark this answers is not in the document]
Comment (micromaomao) [reply cmd/sign1util/main.go:77]: won't fix (no alternative)

File: cmd/attach_receipt/main.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
+// attach_receipt rewrites a COSE_Sign1 envelope's unprotected `receipts`
+// header (label 394), replacing it with either a single raw receipt blob or
+// with the receipts copied from a donor COSE_Sign1 envelope. The protected
+// header, payload, and signature of the input envelope are preserved
+// byte-for-byte by editing the outer CBOR array in place.
```
[the remark this answers is not in the document]
Comment (micromaomao) [reply cmd/attach_receipt/main.go]: we should just get rid of the implementation details from this comment, e.g.

```
// attach_receipt rewrites a COSE_Sign1 envelope's unprotected `receipts`
// header (label 394), replacing it with either a single raw receipt blob or
// with the receipts copied from a donor COSE_Sign1 envelope
```

File: cmd/attach_receipt/main.go [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+// header (label 394), replacing it with either a single raw receipt blob or
+// with the receipts copied from a donor COSE_Sign1 envelope.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/fxamacker/cbor/v2"
```
Comment (maxtropets) [inline cmd/attach_receipt/main.go:11]: I'm not familiar with dependency infra in Go, but just pointing out that it may be unsafe to import an unpinned 3rdparty dependency?

File: cmd/sign1util/ccf_keyfetch.go [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+	if err != nil {
+		return "", fmt.Errorf("issuer %q is not a valid host[:port]: %w", issuer, err)
+	}
+	if u.Scheme != "" || u.User != nil || u.Path != "" || u.RawQuery != "" || u.Fragment != "" {
+		return "", fmt.Errorf("issuer %q must be a plain host[:port]", issuer)
+	}
+	host := u.Hostname()
+	if host == "" {
+		return "", fmt.Errorf("issuer %q has no host", issuer)
+	}
```
Comment (micromaomao) [inline cmd/sign1util/ccf_keyfetch.go]: @Copilot Can we just use a regex like `^([a-zA-Z0-9\-]+\.)+[a-zA-Z0-9\-]+(:\d+)?`, then if that validates, use that as the host, without url.Parse?

File: cmd/sign1util/ccf_keyfetch.go [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+	hostLower := strings.ToLower(host)
+	allowed := false
+	for _, d := range allowedDomains {
+		dl := strings.ToLower(strings.TrimSpace(d))
+		if dl == "" {
+			continue
+		}
```
Comment (micromaomao) [inline cmd/sign1util/ccf_keyfetch.go]: @Copilot why would we ever have this?

File: cmd/sign1util/main.go [hunk 9]
```diff
@@ -1,16 +1,83 @@
...
+	}
+}
+
+func printKeyValue(indent string, k, v interface{}) {
+	fmt.Fprintf(os.Stdout, "%s%v: %s\n", indent, k, formatValue(v))
+}
+
+// combinedJWKSDomains returns the default JWKS domain allow list extended
+// with any additional domains supplied by the user via --allow-jwks-domain.
+func combinedJWKSDomains(extra []string) []string {
```
Comment (micromaomao) [inline cmd/sign1util/main.go]: @Copilot rename this to "extendJWKSDominsAllowList"

**Review by micromaomao — COMMENTED** [review]
@Copilot fix all comments I just made pls

File: cmd/attach_receipt/main.go [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+	return arr, nil
+}
+
+func replaceReceipts(data []byte, receipts []interface{}) ([]byte, error) {
+	msg, hadTag, err := decodeSign1(data)
+	if err != nil {
+		return nil, err
+	}
+	// Delete any existing receipts entry (under either int key type) then set.
+	delete(msg.Unprotected, int64(cosesign1.COSE_Header_Receipts))
```
Comment (maxtropets) [inline cmd/attach_receipt/main.go:99]: Nit. This functionality seems lacking tests, as in to make sure it preserves other key-value pair in the maps, etc. 

In particular, it's worth testing that `delete(uint)` taking place after `delete(int)` actually does smth, or does the `cbor` have it covered already by internally by treating all uint64 that fit into 64 bits as int64s?

Comment (maxtropets) [reply cmd/attach_receipt/main.go:99]: Also, this module seems to have no tests at all?

Comment (micromaomao) [reply cmd/attach_receipt/main.go:99]: This is really a one-off testing tool that I decided to keep in this repo, but is not otherwise a supported (or useful) functionality, so I'm happy with no tests for this for now

File: cmd/attach_receipt/main.go (hunk 6 above)
Comment (micromaomao) [reply cmd/attach_receipt/main.go:11]: I'm not an expert on Go dependency management either, but this is an existing indirect dependency (introduced by [go-cose](https://github.com/veraison/go-cose/blob/main/cbor.go#L8)) that's now directly used in check.go to unmarshal the inclusion proof, and so I guess at least we're not introducing a new third-party dependency.

@copilot+claude-opus-4.7 does the presence of this in go.sum pin it on `go build` runs?

Comment (micromaomao) [reply cmd/attach_receipt/main.go:11]: Copilot didn't realize it need to reply to this, but here is the answer:

<details>
Yes. In Go modules, versions are pinned by go.mod (which lists exact semver versions, including for indirect dependencies once go mod tidy has been run) and go.sum records cryptographic hashes of those exact module versions.

On go build:

The Go toolchain resolves dependencies to the exact versions selected by minimal version selection from go.mod — it does not pick "latest". So the version is pinned.
Every module zip / go.mod it downloads is checked against the hash in go.sum. If the upstream content for that version ever changed, the build fails with a checksum mismatch error.
With GOFLAGS=-mod=readonly (the default outside of commands like go get/go mod tidy), the build will also refuse to silently add or upgrade modules.
So once github.com/fxamacker/cbor/v2 vX.Y.Z is in go.mod + go.sum, go build will always use that exact version with that exact content, whether it's a direct or indirect dependency. Promoting it from indirect to direct (by importing it in attach_receipt/main.go) doesn't change the pinning guarantees — go mod tidy will just drop the // indirect comment next to the existing line.

The only ways the version would change are explicit actions: editing go.mod, running go get ...@version, or go mod tidy picking up a newer version because some other dependency now requires it (still subject to go.sum verification for whatever version ends up selected).
</details>

tl;dr: `go build` would not update the dependency version on its own, it will check that the hash in go.sum matches. But a future `go mod tidy` may bump this dependency. I think this is reasonably acceptable (since we're already using a third-party go-cose anyway)?

File: cmd/attach_receipt/main.go (hunk 10 above)
Comment (micromaomao) [reply cmd/attach_receipt/main.go:99]: (actually in that case maybe this doesn't belong in the main branch? @KenGordon do you have any preference if we keep this attach_receipt thing?)

File: cmd/sign1util/ccf_keyfetch.go [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+// server is passed to verifyCert. The issuer host is validated against
+// allowedDomains before any network request is made.
+func fetchIssuerJWKS(issuer string, allowedDomains []string, verifyCert CertVerifier) (map[string]crypto.PublicKey, error) {
+	host, err := validateIssuerForJWKSFetch(issuer, allowedDomains)
+	if err != nil {
+		return nil, err
+	}
+	reqURL := (&url.URL{Scheme: "https", Host: host, Path: "/jwks"}).String()
+	tlsConfig := &tls.Config{InsecureSkipVerify: true} //nolint:gosec // CCF uses self-signed certs that are supposed to be validated via attestation, and so will never pass the normal verification.
+	if verifyCert != nil {
```
Comment (maxtropets) [inline cmd/sign1util/ccf_keyfetch.go:117]: Will this ever be called with the preceding `tlsConfig := &tls.Config{InsecureSkipVerify: true}`?

File: cmd/sign1util/main.go [hunk 12]
```diff
@@ -33,13 +100,60 @@ func checkCoseSign1(inputFilename string, chainFilename string, didString string
 	}
 
 	fmt.Fprint(os.Stdout, "checkCoseSign1 passed\n")
+
+	// If the envelope carries COSE Receipts, validate each against the CCF
+	// ledger profile. Receipt validation triggers outbound JWKS fetches and
+	// is therefore opt-in via --validate-receipts.
+	var receiptKeys map[string]crypto.PublicKey
+	if validateReceipts && len(unpacked.Receipts) > 0 {
```
Comment (maxtropets) [inline cmd/sign1util/main.go]: If user admits `validateReceipts := true`, shall we then fail if `len(unpacked.Receipts) == 0`?

File: cmd/sign1util/ccf_keyfetch.go (hunk 11 above)
Comment (micromaomao) [reply cmd/sign1util/ccf_keyfetch.go:117]: Yes, the doc says:
> ... If normal verification is disabled (on the client when InsecureSkipVerify is
> set, or on a server when ClientAuth is RequestClientCert or
> RequireAnyClientCert), then this callback will be considered but the
> verifiedChains argument will always be nil. ...

Also tested that returning non-nil error from this function will cause error.

Comment (micromaomao) [reply cmd/sign1util/ccf_keyfetch.go:117]: will add a comment, thanks for raising!

File: cmd/sign1util/main.go (hunk 12 above)
Comment (micromaomao) [reply cmd/sign1util/main.go]: hmm... that's a good point.

However, this application doesn't really force that the receipt has to come from
any particular ledger, and I guess for a proper verification the user probably
cares about that.  So maybe we need a flag like `--require-receipt-from esrp-cts-cp.confidential-ledger.azure.com`.

But then this is still not perfect, since we accept any TLS certificates.  A
real verifier will need to either check the attestation on the CCF, or accept a
TTL (i.e. the trusted ledger key list) signed by some other trusted party.

I can add the `require-receipt-from` flag that takes in the domain if you think
this is sensible - this would be useful as a check tool in the UVM build
pipeline, but at the end of the day the receipt verification here is incomplete
without solving a larger problem, and `sign1util`'s receipt verification can
really only be for debugging / testing.  The actual UVM will implement TTL.

**Comment by micromaomao** [comment]
@copilot+claude-opus-4.7 For `sign1util check`, add a --require-receipt-from flag that takes in the domain, replacing --validate-receipts. This flag will add the provided domain to the JWKs fetching allow list, and will check that at least one receipt in the validated cosesign document has this as the issuer. For any other receipts present, just ignore them. This also means that we can get rid of the --allow-jwks-domain flag from `sign1util check` (but leave it for `sign1util print`).

Note that the receipt issuer matching check should be exact match, not subdomains.

For `sign1util print`, in the help text of `--validate-receipts`, say:
> If any receipts are present, validate any attached transparent receipts, fetching necessary keys from the issuer. This does not enforce any condition on the issuer of the receipts, but only checks that they are valid.

(This is from the discussion https://github.com/microsoft/cosesign1go/pull/38#discussion_r3311131777)

File: pkg/cosesign1/check.go [hunk 13]
```diff
@@ -190,5 +299,285 @@ func UnpackAndValidateCOSE1CertChain(raw []byte) (*UnpackedCoseSign1, error) {
...
+//   - the protected header carries a vds (label 395),
+//   - the payload is detached,
+//   - the unprotected `vdp` header (label 396) contains at least one
+//     inclusion proof (key -1) encoded as a byte string,
+//   - the Merkle root recomputed from each inclusion proof verifies the
+//     receipt's COSE_Sign1 signature, using the public key in `keys` indexed by
+//     r.Kid.
+//   - The data-hash in the receipt matches the expected hash of the signed
+//     statement it is for.
+func (r ParsedCOSEReceipt) Validate(keys map[string]crypto.PublicKey) error {
```
Comment (maxtropets) [inline pkg/cosesign1/check.go:347]: Shall this also perform the CCF receipt's issuer check? I think it should match the one used for key fetching?

Comment (micromaomao) [reply pkg/cosesign1/check.go:347]: Key fetching is outside the responsibility of this function, and in fact we don't pass in any "expected issuer" to this function at all. The caller is expected to first check that `r.Issuer` matches with what they expect (e.g. check that the issuer is listed in the rego policy or something), then fetch the keys based on that issuer (on a production UVM this will be through TTL rather than making https requests), and pass the result of that key fetching here.

File: cmd/sign1util/main.go [hunk 14]
```diff
@@ -33,13 +100,60 @@ func checkCoseSign1(inputFilename string, chainFilename string, didString string
 	}
 
 	fmt.Fprint(os.Stdout, "checkCoseSign1 passed\n")
```
Comment (maxtropets) [inline cmd/sign1util/main.go:119]: I can't find did:x509 resolution from the MST issuer to the COSE key. Isn't it necessary? Or is implemented elsewhere?

Comment (micromaomao) [reply cmd/sign1util/main.go:119]: `sign1util check` will check the did:x509 matches the certificate, but `checkCoseSign1` itself doesn't if didString is not provided (even if it's in the issuer of the cosesign1 document).

(Unrelated to this PR, `UnpackAndValidateCOSE1CertChain` also doesn't validate that the issuer in the cose document, if it is a did, verifies against the cert chain. GCS does do it tho, when validating a fragment.)

**Review by maxtropets — APPROVED** [review]

**Review by KenGordon — APPROVED** [review]
